packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -75,6 +87,15 @@ export type ConsumerHandler<T, U> =
         headers?: any
       ) => Promise<SubscribeResponse>;
     })
+  | (BaseConsumerHandler & {
```

Comment:
The same would be applied here, less type pollution and a more streamlined type.

Reply:
yeah these types look redundant just in general. i'll see what i can do here

Reply:
simplified them a bit and tied them to the main handler types. there might be an opportunity to go even further with conditional types but i don't think it's really necessary